ddtrace/datadog/src/exporter.cpp

```diff
@@ -0,0 +1,500 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+// High-level skip for invalid architectures
+#ifndef __linux__
+#elif __aarch64__
+#elif __i386__
+#else
+
+#include "exporter.hpp"
+
+#include <iostream>
+#include <thread>
+
+using namespace Datadog;
+
+inline ddog_CharSlice to_slice(const std::string_view &str) {
+  return {.ptr = str.data(), .len = str.size()};
+}
+
+DdogProfExporter::DdogProfExporter(std::string_view env,
+                                   std::string_view service,
+                                   std::string_view version, 
+                                   std::string_view url) {
+
+  // Setup the 
+  ddog_Vec_Tag tags = ddog_Vec_Tag_new();
+  add_tag(tags, "language", language);
+  add_tag(tags, "env", env);
+  add_tag(tags, "service", service);
+  add_tag(tags, "version", version);
+  add_tag(tags, "profiler_version", profiler_version);
+
+  ddog_prof_Exporter_NewResult new_exporter = ddog_prof_Exporter_new(
+      to_slice("dd-trace-py"),
+      to_slice(profiler_version),
+      to_slice(family),
+      &tags,
+      ddog_Endpoint_agent(to_slice(url)));
+  ddog_Vec_Tag_drop(tags);
+
+  if (new_exporter.tag == DDOG_PROF_EXPORTER_NEW_RESULT_OK)
+    ptr = new_exporter.ok;
+  else
+    std::cout << "ERROR INITIALIZING LIBDATADOG EXPORTER" << std::endl;
+}
+
+DdogProfExporter::~DdogProfExporter() {
+  ddog_prof_Exporter_drop(ptr);
+}
+
+Uploader::Uploader(std::string_view _env,
+                   std::string_view _service,
+                   std::string_view _version,
+                   std::string_view _url) :
+  env(_env),
+  service(_service),
+  version(_version),
+  url(_url) {
+    ddog_exporter = std::make_unique<DdogProfExporter>(env, service, version, url);
```

Comment:
I like the C++ idiom, though I have trouble wrapping my head around the error management